docs/sql/sql.md

```diff
@@ -0,0 +1,475 @@
+# SQL Support
+
+SpacetimeDB supports two subsets of SQL.
+One for ad hoc queries against the database.
+These can be made through the [cli] or the [http] endpoint.
+Another for subscriptions made through the [websocket] api or [sdk].
+
+## Subscriptions
+
+```ebnf
+SELECT projection FROM relation [ WHERE predicate ]
+```
+
+The subscription language is strictly a query language.
+Its sole purpose is to replicate a subset or view of the database,
+and to **automatically** update that view in realtime as the database changes.
+
+There is no context for manually updating this view.
+Hence data manipulation commands are not supported.
+
+> NOTE: Because subscriptions are evaluated in realtime,
+> performance is critical, and as a result,
+> additional restrictions are applied over ad hoc queries.
+> These restrictions are highlighted below.
+
+### SELECT
+
+```ebnf
+SELECT ( '*' | table '.' '*' )
+```
+
+The `SELECT` clause determines the table that is being subscribed to.
+A subscription query may only return rows from a single table,
+and it must return the entire row.
+Individual column projections are not allowed.
+
+A wildcard projection `*` is allowed when the table is unambiguous,
+but when a query contains multiple table references,
+a qualified wildcard projection `.*` is necessary.
+
+### FROM
+
+```ebnf
+FROM table [ [AS] alias ] [ [INNER] JOIN table [ [AS] alias ] ON predicate ]
+```
+
+While you can only subscribe to rows from a single table,
+you may reference two tables in the `FROM` clause using a `JOIN`.
+A `JOIN` selects all combinations of rows from its input tables,
+and `ON` determines which combinations are considered.
+
+> IMPORTANT: If a `JOIN` is used,
+> an index **must** be defined on the join column of both tables,
+> so that it can be evaluated efficiently.
```

Comment:
Note this applies to subscriptions only. This just repeats what is already specified in the proposal, but I want to highlight it in case it was overlooked.

We could make the same restriction for ad hoc queries. Mainly to avoid performance footguns. Ultimately these potential performance issues should be addressed by auth and/or workload management, but we won't have a complete solution that addresses them for 1.0. So I'm bringing this up here in case there are concerns.

Reply:
I would specify in the docs note that it is for subscriptions only.

For 1.0 we will have proper energy metering as it's required for Mainnet. I don't think we should disallow it for ad hoc queries as it may be required for things like allowing the BitCraft team to debug the data for example.